agent/src/main/java/com/cloud/agent/Agent.java

```diff
@@ -504,6 +504,13 @@ protected void setupStartupCommand(final StartupCommand startup) {
         startup.setGuid(getResourceGuid());
         startup.setResourceName(getResourceName());
         startup.setVersion(getVersion());
+        startup.setArch(getAgentArch());
+    }
+
+    protected String getAgentArch() {
+        final Script command = new Script("/usr/bin/arch", 500, logger);
```

Comment:
Thanks, this LGTM, but on second though the virsh node info returns the arch - perhaps instead of forking out a shell process - we can simply query via libvirt connection?
```
> virsh nodeinfo
CPU model:           x86_64
...
# on rpi5:
rohit@cloudpi:~$ virsh nodeinfo
CPU model:           aarch64
CPU(s):              4
CPU socket(s):       1
Core(s) per socket:  4
Thread(s) per core:  1
NUMA cell(s):        1
Memory size:         8132920 KiB
```